sdk/resourcemanager/Azure.ResourceManager.Core/src/Resources/SubResource.cs

```diff
@@ -1,34 +1,89 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+
 namespace Azure.ResourceManager.Core
 {
+    /// <summary>
+    /// A class representing the a sub resource of ResourceIdentifier.
+    /// </summary>
+    public class SubResource : SubResource<ResourceIdentifier>
+    {
+        /// <summary>
+        /// Initializes an empty instance of <see cref="SubResource"/> for mocking.
+        /// </summary>
+        [InitializationConstructor]
+        protected SubResource() { }
+
+        /// <summary> Initializes a new instance of <see cref="SubResource"/>. </summary>
+        /// <param name="id"> ARM resource Id. </param>
+        [SerializationConstructor]
+        protected internal SubResource(string id) : base(id) { }
+    }
+
     /// <summary>
     /// A class representing a sub-resource that contains only the ID.
     /// </summary>
     [ReferenceType]
-    public partial class SubResource
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("StyleCop.CSharp.MaintainabilityRules", "SA1402:File may only contain a single type", Justification = "Types differ by type argument only")]
+    public partial class SubResource<TIdentifier> : IEquatable<SubResource<TIdentifier>>, IEquatable<string>,
+        IComparable<SubResource<TIdentifier>>, IComparable<string> where TIdentifier : ResourceIdentifier
     {
         /// <summary>
-        /// Initializes an empty instance of <see cref="SubResource"/> for mocking.
+        /// Initializes an empty instance of <see cref="SubResource{TIdentifier}"/> for mocking.
         /// </summary>
         [InitializationConstructor]
-        public SubResource()
-        {
-        }
+        protected SubResource() { }
 
-        /// <summary> Initializes a new instance of SubResource. </summary>
+        /// <summary> Initializes a new instance of <see cref="SubResource{TIdentifier}"/>. </summary>
         /// <param name="id"> ARM resource Id. </param>
         [SerializationConstructor]
         protected internal SubResource(string id)
         {
-            Id = id;
+            Id = ResourceIdentifier.Create(id) as TIdentifier;
```

Comment:
pretty sure you don't have to call create here the implicit will handle it you can do a straight assignment
Id = id;

Reply:
me thinks i do 
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0266	Cannot implicitly convert type 'string' to 'TIdentifier'. An explicit conversion exists (are you missing a cast?)	Azure.ResourceManager.Core (netstandard2.0)	S:\src\azure-sdk-for-net\sdk\resourcemanager\Azure.ResourceManager.Core\src\Resources\SubResource.cs	50	Active